clinical-domain-agent/src/test/java/care/smith/fts/cda/impl/EverythingDataSelectorIT.java

```diff
@@ -0,0 +1,144 @@
+    wireMock = wireMockRuntime.getWireMock();
+    client = clientFactory.create(clientConfig(wireMockRuntime));
+    var common = new DataSelector.Config(false, null);
+    EverythingDataSelectorIT.meterRegistry = meterRegistry;
+    dataSelector =
+        new EverythingDataSelector(
+            common, client, pidResolver, EverythingDataSelectorIT.meterRegistry, PAGE_SIZE);
+
+    var consentedPolicies = new ConsentedPolicies();
+    consentedPolicies.put("pol", new Period(ZonedDateTime.now(), ZonedDateTime.now().plusYears(5)));
+    consentedPatient = new ConsentedPatient(PATIENT_ID, consentedPolicies);
+  }
+
+  private static MappingBuilder getBuilder(boolean withConsent) {
+    if (withConsent) {
+      var period = consentedPatient.consentedPolicies().maxConsentedPeriod().get();
+      var start = period.start().format(ISO_LOCAL_DATE.withZone(ZoneId.systemDefault()));
+      var end = period.end().format(ISO_LOCAL_DATE.withZone(ZoneId.systemDefault()));
+      return get("/Patient/%s/$everything?_count=%s&start=%s&end=%s"
+              .formatted(PATIENT_ID, PAGE_SIZE, start, end))
+          .withHeader(ACCEPT, equalTo(APPLICATION_FHIR_JSON));
+    } else {
+      return get("/Patient/%s/$everything?_count=%s".formatted(PATIENT_ID, PAGE_SIZE))
+          .withHeader(ACCEPT, equalTo(APPLICATION_FHIR_JSON));
+    }
+  }
+
+  @Override
+  protected Stream<TestStep<?>> createTestSteps() {
+    return Stream.of(
+        new TestStep<ConsentedPatientBundle>() {
+          @Override
+          public MappingBuilder getBuilder() {
+            return EverythingDataSelectorIT.getBuilder(true);
+          }
+
+          @Override
+          public Flux<ConsentedPatientBundle> executeStep() {
+
+            return dataSelector.select(consentedPatient);
+          }
+
+          @Override
+          public String acceptedContentType() {
+            return APPLICATION_FHIR_JSON_VALUE;
+          }
+        });
+  }
+
+  @Test
+  void noConsentErrors() {
+    create(dataSelector.select(new ConsentedPatient(PATIENT_ID))).expectError().verify();
+  }
+
+  @Test
+  void noConsentSucceedsIfConsentIgnored() {
+    DataSelector.Config common = new DataSelector.Config(true, null);
+    var dataSelector = new EverythingDataSelector(common, client, pidResolver, meterRegistry, 500);
+
+    wireMock.register(
+        getBuilder(false)
+            .willReturn(
+                ok().withHeader("Content-Type", APPLICATION_FHIR_JSON_VALUE)
+                    .withBody(fhirResourceToString(new Bundle()))));
+
+    create(dataSelector.select(consentedPatient))
+        .assertNext(b -> assertThat(b.consentedPatient().id()).isEqualTo(PATIENT_ID))
+        .verifyComplete();
+  }
+
+  @Test
+  void selectionSucceeds() {
+    wireMock.register(
+        getBuilder(true)
+            .willReturn(
+                ok().withHeader("Content-Type", APPLICATION_FHIR_JSON_VALUE)
+                    .withBody(fhirResourceToString(new Bundle()))));
```

Comment:
We have fhirResponse in MockServerUtil, it does this. We could add one without status code, defaulting to OK.